src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1524,10 +1524,8 @@ public sealed override bool IsTupleCompatible(out int tupleCardinality)
             return false;
         }
 
-        // PROTOTYPE: Should be abstract.
-        internal virtual NamedTypeSymbol AsNativeInt(bool asNativeInt) => this;
+        internal virtual NamedTypeSymbol AsNativeInt(bool asNativeInt) => throw ExceptionUtilities.Unreachable;
```

Comment:
>AsNativeInt [](start = 41, length = 11)

When is this API is called?

Reply:
`NamedTypeSymbol.AsNativeInt(bool)` is used when: binding `nint` or `nuint` from source; importing a `[NativeInteger] System.[U]IntPtr` from metadata (in a separate PR); and when constructing signatures for built-in operators for `nint` and `nuint`.

---
In reply to: [370304665](https://github.com/dotnet/roslyn/pull/40994#discussion_r370304665) [](ancestors = 370304665)